modules/Users/models/Record.php

```diff
@@ -1042,4 +1043,59 @@ public function getFavouritesUsers()
 		}
 		return $favouriteUsers;
 	}
+
+	/**
+	 * Update all users labels
+	 *
+	 * @return void
+	 */
+	public static function updateAllLabels(): void
+	{
+		$usersId = (new \App\Db\Query())->select(['id'])->from('vtiger_users')->column();
+		foreach ($usersId as $userId) {
+			(new \App\BatchMethod([
+				'method' => static::class . '::updateLabelUser',
+				'params' => [$userId]
+			]))->save();
+		}
+	}
+
+	/**
+	 * Update given userId label.
+	 *
+	 * @param int $userId
+	 *
+	 * @return void
+	 */
+	public static function updateLabelUser(int $userId): void
+	{
+		if (\App\User::isExists($userId)) {
+			$userRecordModel = self::getInstanceById($userId, 'Users');
+			$userRecordModel->updateLabel();
+		}
+	}
```

Comment:
one static function
and maybe App\User or even Users_Model would be a better place

Reply:
App\User::updateAllLabels
App\User::updateUserLabel